src/patch-events.js

```diff
@@ -519,27 +520,43 @@ function activateFocusEventOverrides() {
   }
 }
 
+const EventPatchesDescriptors = utils.getOwnPropertyDescriptors(EventPatches);
+
+const SHADY_PROTO = '__shady_patchedProto';
+const SHADY_SOURCE_PROTO = '__shady_sourceProto';
+
 function patchEvent(event) {
   event['__target'] = event.target;
   event.__relatedTarget = event.relatedTarget;
-  // patch event prototype if we can
+  // attempt to patch prototype (via cache)
   if (utils.settings.hasDescriptors) {
-    utils.patchPrototype(event, eventMixin);
+    const proto = Object.getPrototypeOf(event);
+    if (!Object.hasOwnProperty(proto, SHADY_PROTO)) {
+      const patchedProto = Object.create(proto);
+      patchedProto[SHADY_SOURCE_PROTO] = proto;
+      utils.patchProperties(patchedProto, EventPatchesDescriptors, true);
+      proto[SHADY_PROTO] = patchedProto;
+    }
+    event.__proto__ = proto[SHADY_PROTO];
   // and fallback to patching instance
   } else {
-    utils.extend(event, eventMixin);
+    utils.patchProperties(event, EventPatchesDescriptors, true);
   }
 }
 
-let PatchedEvent = mixinComposedFlag(window.Event);
-let PatchedCustomEvent = mixinComposedFlag(window.CustomEvent);
-let PatchedMouseEvent = mixinComposedFlag(window.MouseEvent);
+let PatchedEvent = mixinComposedFlag(Event);
+let PatchedCustomEvent = mixinComposedFlag(CustomEvent);
+let PatchedMouseEvent = mixinComposedFlag(MouseEvent);
 
 export function patchEvents() {
+  activateFocusEventOverrides();
```

Comment:
Ideally `noPatch` mode would not patch the `Event` constructors, and instead require users to opt-in to creating patched events (e.g. `new ShadyDOM.wrap.Event('...', {...})`), so as to not add the patching tax to events either.

Reply:
Elected not to do this to avoid confusion.